### PR TITLE
Fix bugs in SQL`ScriptScanner` with big String literals and PostgreSQL identifiers (as introduced by #7646)

### PR DESCRIPTION
**Pull request by inponomarev**
"This never happened before, and here we go again": unfortunately #7646 introduced not only proper bug fixes, but some new bugs as well, reproducible on some of our test SQL scripts.

This PR introduces 3 changes:

1. (⚠️ The biggest problem). It seems that Java's regexp `"(\\"|[^"])*"`, though working well with strings of a reasonable size, tends to yield `StackOverflowError` when literals are big (10k simbols and more). The workaround is the same as with comments: use finite-state machine to match the literal, not the regexp.
2.  According to [PostgreSQL docs](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS), dollar sign can be a part of an identifier  (non-standard behaviour for SQL, as they themselves admit), and it must be handled so that `this$is$a$valid$postgreSQL$identifier` is not mixed with dollar-quoted strings
3. `m.find(offset) && m.start() == offset` was suboptimal (we are only interested in cases when the regexp matches exactly starting from the offset and there is no reason in finding it somewhere far away in the script).

Existing tests are not modified, new tests added.

### CODE REVIEW

**Review by kiview — APPROVED** [review]

File: modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.Test;
+
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ScriptScannerTest {
+    @Test
+    public void testHugeStringLiteral() {
+        String script = "/* a comment */    \"" +  StringUtils.repeat('~', 10000) + "\";";
```
Comment (kiview) [inline modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java:13]: This line makes checkstyle fail. @eddumelendez do you know how to mark it as an exception?

**Review by kiview — CHANGES_REQUESTED** [review]

File: modules/database-commons/src/main/java/org/testcontainers/ext/ScriptScanner.java [hunk 2]
```diff
@@ -99,6 +96,27 @@ private boolean matchesMultilineComment() {
         return false;
     }
 
+
```
Comment (eddumelendez) [inline modules/database-commons/src/main/java/org/testcontainers/ext/ScriptScanner.java:99]:
```suggestion
```

File: modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Test;
+
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ScriptScannerTest {
+    @Test
```
Comment (eddumelendez) [inline modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java:11]:
```suggestion

    @Test
```

**Review by eddumelendez — CHANGES_REQUESTED** [review]

File: modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ScriptScannerTest {
+
+    @Test
+    public void testHugeStringLiteral() {
+        String script = "/* a comment */    \"" +  StringUtils.repeat('~', 10000) + "\";";
```
Comment (eddumelendez) [inline modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java:14]:
```suggestion
        String script = "/* a comment */    \"" + StringUtils.repeat('~', 10000) + "\";";
```

File: modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java (hunk 1 above)
Comment (eddumelendez) [reply modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java]: it was a double space after first `+`

File: modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java (hunk 4 above)
Comment (kiview) [reply modules/database-commons/src/test/java/org/testcontainers/ext/ScriptScannerTest.java]: Oh it was this space, not the one in the String 😬

**Review by kiview — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks for your contribution, @inponomarev !

**Pull request merged**